src/pages/SignUp/SignUpForm.js

```diff
@@ -0,0 +1,183 @@
+  const [isValidate, setIsValidate] = useState(false);
+
+  const regName = /^[a-zA-Z0-9]{4,20}$/;
+  const regEmail = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{3}$/;
+  const regPassword =
+    /^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,}$/;
+
+  // const navigate = useNavigate();
+  // const goToMain = () => {
+  //   navigate('/main-soojin');
+  // };
+
+  const handleInput = e => {
+    const { name, value } = e.target;
+    setInputValues({
+      ...inputValues,
+      [name]: value,
+    });
+  };
+
+  const testValidation = () => {
+    regName.test(inputValues.name) &&
+    regEmail.test(inputValues.email) &&
+    regPassword.test(inputValues.password) &&
+    inputValues.repeatPassword === inputValues.password &&
+    isChecked === true
+      ? setIsValidate(true)
+      : setIsValidate(false);
+  };
+
+  const clickButton = () => {
+    // console.log(inputValues, isChecked);
+    fetch('http://10.58.2.82:8000/users/signup', {
+      method: 'POST',
+      body: JSON.stringify({
+        name: inputValues.name,
+        email: inputValues.email,
+        password: inputValues.password,
+        // has_agreed: isChecked,
+        // name: 'wwww',
+        // email: 'id@example.com',
+        // password: 'asdf1234!',
+      }),
+    })
+      .then(response => response.json())
+      .then(result => console.log(result));
+  };
+
+  useEffect(() => {
+    // console.log(inputData);
+    testValidation();
+  }, [inputValues, isChecked]);
+
+  // console.log(inputValues);
+  // console.log(isChecked);
+  // console.log(isValidate);
+
+  return (
+    <SignUpBody>
+      <FormLayout>
+        <Heading>{title}</Heading>
+        {inputData.map((input, idx) => (
+          <Input
+            key={idx}
+            name={input.name}
+            text={input.text}
+            type={input.type}
+            placeholder={input.placeholder}
+            handleInput={handleInput}
+          />
+        ))}
+
+        <AgreementWrap>
+          <AgreementCheckbox
+            type="checkbox"
+            name="checkbox"
+            onClick={() => {
+              setIsChecked(!isChecked);
+            }}
+          />
+          <AgreementH1>회원 가입 및 운영약관 동의 (필수)</AgreementH1>
+        </AgreementWrap>
+
+        <ButtonWrap>
+          <SignUpButton disabled={!isValidate} onClick={clickButton}>
+            동의하고 가입하기
+          </SignUpButton>
+        </ButtonWrap>
+
+        <SwitchWrap>
+          <SwitchLink>이미 아이디가 있으신가요?</SwitchLink>
+        </SwitchWrap>
+      </FormLayout>
+    </SignUpBody>
+  );
+};
+
+export default SignUpForm;
+
+const SignUpBody = styled.div`
+  height: 100vh;
+  position: relative;
+`;
+
+const Heading = styled.h1`
+  font-size: 24px;
+  font-weight: bold;
+`;
+
+const AgreementWrap = styled.div`
+  display: flex;
+  padding: 12px;
+  width: 320px;
+  border: ${props => props.theme.border};
+  margin: auto;
+  margin-top: 12px;
+  margin-bottom: 12px;
```

Comment:
```suggestion
  margin: 12px auto;
```

해당 css 요소는 한줄로 줄일수 있습니다!